cpp/daal/src/externals/service_profiler.cpp

```diff
@@ -16,50 +16,307 @@
 
-ProfilerTask::ProfilerTask(const char * taskName) : _taskName(taskName)
+static volatile int daal_verbose_val = -1;
+
+void print_header()
 {
-    _handle = __itt_string_handle_create(taskName);
+    daal::services::LibraryVersionInfo ver;
 
-    __itt_task_begin(Profiler::getDomain(), __itt_null, __itt_null, _handle);
+    std::cout << "Major version:          " << ver.majorVersion << std::endl;
+    std::cout << "Minor version:          " << ver.minorVersion << std::endl;
+    std::cout << "Update version:         " << ver.updateVersion << std::endl;
+    std::cout << "Product status:         " << ver.productStatus << std::endl;
+    std::cout << "Build:                  " << ver.build << std::endl;
+    std::cout << "Build revision:         " << ver.build_rev << std::endl;
+    std::cout << "Name:                   " << ver.name << std::endl;
+    std::cout << "Processor optimization: " << ver.processor << std::endl;
+    std::cout << std::endl;
 }
 
-ProfilerTask::~ProfilerTask()
+static void set_verbose_from_env(void)
 {
-    Profiler::endTask(_taskName);
+    // Every time check that the env is not changed
+    // static volatile int read_done = 0;
+    // if (read_done) return;
+
+    const char * verbose_str = std::getenv("ONEDAL_VERBOSE");
+    int newval               = 0;
+    if (verbose_str)
+    {
+        newval = std::atoi(verbose_str);
+        if (newval < 0 || newval > 5)
+        {
+            newval = 0;
+        }
+        else
+        {
+            print_header();
+        }
+    }
+
+    daal_verbose_val = newval;
+    // read_done        = 1;
 }
 
-ProfilerTask Profiler::startTask(const char * taskName)
+std::string format_time_for_output(std::uint64_t time_ns)
 {
-    return ProfilerTask(taskName);
+    std::ostringstream out;
+    double time = static_cast<double>(time_ns);
+
+    if (time <= 0)
+    {
+        out << "0.00s";
+    }
+    else
+    {
+        if (time > 1e9)
+        {
+            out << std::fixed << std::setprecision(2) << time / 1e9 << "s";
+        }
+        else if (time > 1e6)
+        {
+            out << std::fixed << std::setprecision(2) << time / 1e6 << "ms";
+        }
+        else if (time > 1e3)
+        {
+            out << std::fixed << std::setprecision(2) << time / 1e3 << "us";
+        }
+        else
+        {
+            out << static_cast<std::uint64_t>(time) << "ns";
+        }
+    }
+
+    return out.str();
 }
 
-void Profiler::endTask(const char * taskName)
+/**
+* Returns the pointer to variable that holds oneDAL verbose mode information (enabled/disabled)
+*
+*  @returns pointer to mode
+*                      0/empty disabled
+*                      1 enabled only with logger
+*                      2 enabled only with tracer
+*                      3 enabled only with analyzer
+*                      4 enabled with logger tracer and analyzer
+*                      5 enabled with logger tracer and analyzer with service functions
+*/
+int * daal_verbose_mode()
```

Comment:
It's not obvious why it is returned by pointer here. Can you please explain?

Reply:
it has been fixed :)